devito/symbolics/printer.py

```diff
@@ -159,8 +169,17 @@ def _print_Abs(self, expr):
         if isinstance(self.compiler, AOMPCompiler):
             return "fabs(%s)" % self._print(expr.args[0])
         # Check if argument is an integer
-        func = "abs" if has_integer_args(*expr.args[0].args) else "fabs"
-        return "%s(%s)" % (func, self._print(expr.args[0]))
+        if has_integer_args(*expr.args[0].args):
+            func = "abs"
+        elif self.single_prec(expr):
+            func = "fabsf"
+        elif any([isinstance(a, Real) for a in expr.args[0].args]):
```

Comment:
From the look of it this seems un-necessary. You just need

```
If self.single_prec(expr) or self.single_prec():
    fabsf
else:
    # not integer or f32, default 
    fabs
```

Reply:
Not so, this is the example that I shared on slack, but will repeat here for reference:
```python
>>> i1 = Dimension(name="i1")
>>> sympy_dtype(i1 - 0.5)
<class 'numpy.int32'>
```
In the second part of the `or` in your example `self.single_prec()` will always return `True`, regardless of the types of the expression (unless the default is changed).

This code is a bit horrible, but ensures that the "default" is only used if there is a floating point number whose type cannot be determined. Otherwise strictly use the correct function call for `float` or `double`.

Reply:
Ah, but my test is wrong anyway...

Reply:
I'm going to merge this, but perhaps in a subsequent PR you could attach a comment to that `if`